packages/jellyfish-api-core/__tests__/category/wallet/getBalances.test.ts

```diff
@@ -0,0 +1,90 @@
+import { MasterNodeRegTestContainer, RegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { BigNumber } from '../../../src'
+import { WalletFlag, WalletBalances } from '../../../src/category/wallet'
+
+// TODO(aikchun): Add behavior tests for untrusted_pending, immature, used. Currently unable to do multi-node testing
+describe('getBalances on masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should getBalances', async () => {
+    const balances: WalletBalances = await client.wallet.getBalances()
+    expect(BigNumber.isBigNumber(balances.mine.trusted)).toStrictEqual(true)
+    expect(BigNumber.isBigNumber(balances.mine.untrusted_pending)).toStrictEqual(true)
+    expect(BigNumber.isBigNumber(balances.mine.immature)).toStrictEqual(true)
+    expect(typeof balances.mine.used).toStrictEqual('undefined')
+
+    expect(typeof balances.watchonly).toStrictEqual('undefined')
+  })
+
+  it('should have used in getBalances when wallet is set to avoid_reuse', async () => {
+    await client.wallet.setWalletFlag(WalletFlag.AVOID_REUSE)
```

Comment:
setting this will influence other tests creating a side effect? maybe you should use another container within another describe block.

Reply:
Created a separate describe block for this test.